pandas/core/internals.py

```diff
@@ -1916,6 +1938,12 @@ def _slice(self, slicer):
 
         return self.values[slicer]
 
+    def setitem(self, indexer, value, mgr=None):
```

Comment:
Want to make sure this change isn't overlooked.

I haven't completely nailed down the type of `indexer`, but in my testing it seems like it may or may not be a tuple depending on whether or not the frame is mixed?

Either way, I *think* that since ExtensionBlocks are always 1-D, we want to ignore the second component of the tuple before passing to `self.values.__setitem__`.

Reply:
>  I suppose this is because it checks dtypes and if needed upcasts to common type?

that's the majority of the complexity, yes. There's a bit around the shape of the indexer / target / value that *may* not matter for 1-D ExtensionBlocks, but I'm not confident that it isn't required.

Since `CategoricalBlock` is getting the new `ExtensionBlock.setitem`, this has the potential to actually break things. I've removed it for now, and will start a followup PR attempting to implement it correctly.